src/main/java/org/opensearch/neuralsearch/processor/normalization/MinMaxScoreNormalizationTechnique.java

```diff
@@ -167,22 +224,197 @@ private float[] getMinScores(final List<CompoundTopDocs> queryTopDocs, final int
+     * Get lower bounds from input parameters
+     * @param params user provided input parameters for this technique
+     * @return optional list of lower bounds. Can be empty in case lower bounds are not provided
+     */
+    private Optional<List<Pair<LowerBound.Mode, Float>>> getLowerBounds(final Map<String, Object> params) {
+        // validate that the input parameters are in correct format
+        if (Objects.isNull(params) || params.containsKey(PARAM_NAME_LOWER_BOUNDS) == false) {
+            return Optional.empty();
+        }
+
+        Object lowerBoundsObj = params.get(PARAM_NAME_LOWER_BOUNDS);
+        if (lowerBoundsObj instanceof List<?> == false) {
+            throw new IllegalArgumentException("lower_bounds must be a List");
+        }
+        List<?> lowerBoundsParams = (List<?>) lowerBoundsObj;
+        // number of lower bounds must match the number of sub-queries in a hybrid query
+        if (lowerBoundsParams.size() > MAX_NUMBER_OF_SUB_QUERIES) {
+            throw new IllegalArgumentException(
+                String.format(
+                    Locale.ROOT,
+                    "lower_bounds size %d should be less than or equal to %d",
+                    lowerBoundsParams.size(),
+                    MAX_NUMBER_OF_SUB_QUERIES
+                )
+            );
+        }
+        // parse each lower bound item and put all items in a list
+        List<Pair<LowerBound.Mode, Float>> lowerBounds = lowerBoundsParams.stream().map(this::parseLowerBound).collect(Collectors.toList());
+
+        return Optional.of(lowerBounds);
+    }
+
+    @SuppressWarnings("unchecked")
+    /**
+     * Parse each lower bound item and return a pair of mode and min score
+     * @param boundObj lower bound item provided by the client
+     * @return a single pair of mode and min score
+     */
+    private Pair<LowerBound.Mode, Float> parseLowerBound(Object boundObj) {
+        if ((boundObj instanceof Map) == false) {
+            throw new IllegalArgumentException("each lower bound must be a map");
+        }
+
+        Map<String, Object> lowerBound = (Map<String, Object>) boundObj;
+
+        String lowerBoundModeValue = Objects.toString(lowerBound.get(PARAM_NAME_LOWER_BOUND_MODE), "");
+        LowerBound.Mode mode = LowerBound.Mode.fromString(lowerBoundModeValue);
+        float minScore = extractAndValidateMinScore(lowerBound);
+
+        return ImmutablePair.of(mode, minScore);
+    }
+
+    private float extractAndValidateMinScore(Map<String, Object> lowerBound) {
+        Object minScoreObj = lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE);
+        if (minScoreObj == null) {
+            return LowerBound.DEFAULT_LOWER_BOUND_SCORE;
+        }
+        try {
+            float minScore = LowerBound.DEFAULT_LOWER_BOUND_SCORE;
+            if (Objects.nonNull(lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE))) {
+                minScore = Float.parseFloat(String.valueOf(lowerBound.get(PARAM_NAME_LOWER_BOUND_MIN_SCORE)));
```

Comment:
Should we also check if the provided min score is float if not throw an error?

Reply:
this is checked in the wrapping try/catch, we do following exception in such case:
```
throw new IllegalArgumentException("invalid format for min_score: must be a valid float value", e);
```